protobuf/source_reflection.go

```diff
@@ -32,8 +32,7 @@ type ReflectionArgs struct {
 // NewDescriptorProviderReflection returns a DescriptorProvider that reaches
 // out to a reflection server to access file descriptors.
 func NewDescriptorProviderReflection(args ReflectionArgs) (DescriptorProvider, error) {
-	r, deregisterScheme := GenerateAndRegisterManualResolver()
-	defer deregisterScheme()
+	r := GenerateAndRegisterManualResolver()
```

Comment:
We're creating potential memory leak here.
Let's discuss potential solution with a custom resolver, i.e. to register only one resolver, and then reuse it for different services.